mvm/src/mvm.rs

```diff
@@ -144,10 +156,22 @@ where
                             .and_then(|_| session.finish())
                     })
             });
-        self.handle_vm_result(cost_strategy, gas, result)
+
+        if dry_run {
+            // clear cache if it's dry_run?
+            self.clear();
```

Comment:
You don't need to clear the cache on dry run mode. Because cache contains only modules, structs, and functions.
This data is taken only from the chain. The module from ModuleTx will not dirty the cache. But the VM will be forced to reload stdlib and other modules after the dry run.